featuretools/synthesis/dfs.py

```diff
@@ -252,33 +250,18 @@ def dfs(entities=None,
     if features_only:
         return features
 
-    if isinstance(cutoff_time, pd.DataFrame):
-        feature_matrix = calculate_feature_matrix(features,
-                                                  entityset=entityset,
-                                                  cutoff_time=cutoff_time,
-                                                  training_window=training_window,
-                                                  approximate=approximate,
-                                                  cutoff_time_in_index=cutoff_time_in_index,
-                                                  save_progress=save_progress,
-                                                  chunk_size=chunk_size,
-                                                  n_jobs=n_jobs,
-                                                  dask_kwargs=dask_kwargs,
-                                                  verbose=verbose,
-                                                  progress_callback=progress_callback,
-                                                  include_cutoff_time=include_cutoff_time)
-    else:
-        feature_matrix = calculate_feature_matrix(features,
-                                                  entityset=entityset,
-                                                  cutoff_time=cutoff_time,
-                                                  instance_ids=instance_ids,
-                                                  training_window=training_window,
-                                                  approximate=approximate,
-                                                  cutoff_time_in_index=cutoff_time_in_index,
-                                                  save_progress=save_progress,
-                                                  chunk_size=chunk_size,
-                                                  n_jobs=n_jobs,
-                                                  dask_kwargs=dask_kwargs,
-                                                  verbose=verbose,
-                                                  progress_callback=progress_callback,
-                                                  include_cutoff_time=include_cutoff_time)
+    feature_matrix = calculate_feature_matrix(features,
```

Comment:
Do we have a test that validates that this works? i.e. one that passes in both a pandas `cutoff_time` df as well as some a list of instance ids which seems to be what the old check was avoiding.

Reply:
I don't see any existing tests that cover that exact situation, but I tested it out and if you supply both a cutoff time and df and a list of instance ids, the instance ids list is ignored as indicated in the docstring. 

I do think it would be good to add a warning though to notify the user that the instance ids are not being used, and I'll make that update.

Reply:
Added a warning and new test case to cover passing both a cutoff time df and instance ids.